query/src/main/java/org/infinispan/query/impl/SearchManagerImpl.java

```diff
@@ -148,8 +146,7 @@ public void purge(Class<?> entityType) {
 
    @Override
    public <T> T unwrap(Class<T> cls) {
-      if (SearchIntegrator.class.isAssignableFrom(cls) || ExtendedSearchIntegrator.class.isAssignableFrom(cls)
-            || SearchFactoryState.class.isAssignableFrom(cls)) {
+      if (SearchIntegrator.class.isAssignableFrom(cls)) {
```

Comment:
Could we keep this around?
I don't see why we should prevent that: there will always be need for quick experiments which can't need to wait for us to expose proper APIs for everything.